application-diagram-api/src/main/java/com/xwiki/diagram/internal/handlers/GetDiagramLinksHandler.java

```diff
@@ -23,30 +23,43 @@
 import java.util.List;
 import java.util.stream.Collectors;
 
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
 import org.xml.sax.Attributes;
 import org.xml.sax.SAXException;
 import org.xml.sax.helpers.DefaultHandler;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.DocumentReferenceResolver;
 
 /**
- * Parses the XML of a diagram and gets the links to wiki pages that are included.
+ * Parses the XML of a diagram and gets the links to wiki pages that are linked.
  * 
  * @version $Id$
  * @since 1.13
  */
+@Component(roles = GetDiagramLinksHandler.class)
+@Singleton
 public class GetDiagramLinksHandler extends DefaultHandler
 {
+    @Inject
     private DiagramLinkHandler linkHandler;
 
+    @Inject
+    @Named("current")
```

Comment:
I'm worried about resolving using the current document reference. See my previous comment. I would feel more save if the references from the diagram content would be resolved relative to the diagram reference.